configure.ac

```diff
@@ -868,6 +868,8 @@ AS_CASE([$host],
     [arch=arm; model=armv5te; system=linux_eabi],
   [armv5*-*-linux-gnueabi],
     [arch=arm; model=armv5; system=linux_eabi],
+  [arm*-*-linux-gnueabihf],
+    [arch=arm; system=linux_eabihf],
```

Comment:
After thinking more about this, I think it's wrong to include this change in this PR. The other part is fixing a clear bug, so we want it in our bugfix release. This is arguably a feature (it seems to be adding support for a different arm variant, if I understand [this page](https://wiki.debian.org/ArmHardFloatPort) correctly), our ARM guy (Xavier) is on holidays, and I suspect that the change goes against the current policies of "don't support architectures we don't have CI testing enabled on".

Would you remove it from the patch?

Reply:
Actually, I went through this one - I don't think that's true, it's adding a missing fall-through case which is already there for non-hf ARM.

Reply:
IIUC it's more that we support pretty much any ARM and take *advantage* of specifics. Is that right @mshinwell, in Xavier's absence?

Reply:
I don't know much about arm or configure, so I would be happy to turn out wrong on this one.

(@dra27, semi-relatedly: I guess we will eventually need your Approval to merge the PR.)

Reply:
@glondu small ping: could you clarify the nature of this change?

Reply:
@gasche Without this change, the native compiler support is [not correctly detected on armhf](https://buildd.debian.org/status/fetch.php?pkg=ocaml&arch=armhf&ver=4.08.0-1&stamp=1563914129&raw=0), whereas it used to work with 4.05.0. With this change, the native compiler is [compiled and passes all the tests](https://buildd.debian.org/status/fetch.php?pkg=ocaml&arch=armhf&ver=4.08.0-2&stamp=1563964064&raw=0).

Reply:
I'm convinced, thanks.

Reply:
I'm OK with this change on the basis that it just adds a fallback case.  
At some point it will be healthy to simplify the ARM code generator by removing support for ARM variants that no longer exist, e.g. anything older than ARMv6, perhaps even ARMv7.